### DISCUSSION TITLE
Foliate doesn't open in WSL

### DISCUSSION THREAD

**Discussion by nixl0** [johnfactotum/foliate#951] [General]
I've installed Foliate through Flatpak on WSL with Ubuntu 20.04 distro, but it doesn't open. Something like gedit works fine.

Shows this set of messages instead, after which nothing happens:
```
nlx@hewlettpackard:~$ flatpak run com.github.johnfactotum.Foliate
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED
Can't find bus: Cannot autolaunch D-Bus without X11 $DISPLAY
Can't find bus: Cannot autolaunch D-Bus without X11 $DISPLAY

(com.github.johnfactotum.Foliate:2): Gtk-WARNING **: 21:22:52.718: cannot open display:
nlx@hewlettpackard:~$
```

**Comment by MartinJindra** [comment]
Believe this is more a problem with WSL.

**Comment by lelinhtinh** [comment]
Now, Foliate is working perfectly in new WSL version. Follow [this tutorial](https://docs.microsoft.com/windows/wsl/install), or the shortened steps below.

**Prerequisites**: _You must be running Windows 10 version 2004 and higher (Build 19041 and higher) or Windows 11._

1. Enter this command in an administrator PowerShell or Windows Command Prompt:
    ```shell
    wsl --install -d Ubuntu
    ```
1. Then, restarting your machine.
1. Open Ubuntu, set up your Linux username and password.
1. Update:
    ```bash
    sudo apt update && sudo apt upgrade
    ```
1. Install Foliate via PPA:
    ```bash
    sudo add-apt-repository ppa:apandada1/foliate
    sudo apt update
    sudo apt install foliate
    ```
1. Open Foliate from a shortcut via the Start Menu.
![image](https://user-images.githubusercontent.com/5460915/173205384-e1ad520f-9c3d-4652-9f52-7766da312e74.png)

**Comment by gvenusleo** [comment]
I had the same problem, and I've solved it now. This is because of a lack of a driver for vGPU. To run Linux GUI apps, you should first install the driver matching your system. Just follow this [tutorial](https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps).
